src/main/java/seedu/address/model/project/TodoList.java

```diff
@@ -69,7 +69,12 @@ public void deleteTodo(Integer i) {
      */
     public void markAsDone(Integer i) {
         requireNonNull(i);
-        this.todos.get(i).markAsDone();
+        CompletableTodo todo = todos.get(i);
+        todo.markAsDone();
+
+        // Force observable list to update
+        this.todos.remove(todo);
+        this.todos.add(i, todo);
```

Comment:
Can the Set(i,deadline) method be used instead. This way the list would not get reordered unnecessarily.

Reply:
good call